File: 0875-koko-eating-bananas/0875-koko-eating-bananas.c
int minEatingSpeed(int* piles, int pilesSize, int h) {
    int low=1;
    int high=INT_MIN;
    for(int i=0;i<pilesSize;i++)
    {
        if(piles[i]>high)
            high=piles[i];
    }
    while(low<high)
    {
        int mid=low + (high-low)/2;
        int TotalHrs=0;
        for(int i=0;i<pilesSize;i++)
        {
            int divi=piles[i]+mid-1;
            TotalHrs+=divi/mid;
        }
        if(TotalHrs<=h)
            high=mid;
        else
            low=mid+1;
    }
    return low;
}

File: 0875-koko-eating-bananas/0875-koko-eating-bananas.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        low = 1
        high = piles[0]
        for i in piles:
            if(i > high):
                high = i
        
        while(low < high):
            mid = low + (high - low) // 2
            TotalHrs = 0
            for i in piles:
                TotalHrs += (i + mid - 1) // mid

                if(TotalHrs > h):
                    break
            
            if(TotalHrs <= h):
                high = mid
            else:
                low = mid+1
        return low